moveit_core/trajectory_processing/src/iterative_torque_limit_parameterization.cpp

```diff
@@ -45,25 +45,28 @@ constexpr double DEFAULT_VELOCITY_LIMIT = 1.0;
 constexpr double DEFAULT_ACCELERATION_LIMIT = 1.0;
 }
 
-IterativeTorqueLimitParameterization::IterativeTorqueLimitParameterization(const double path_tolerance,
-                                                                           const double resample_dt,
-                                                                           const double min_angle_change)
+IterativeTorqueLimitParameterization::IterativeTorqueLimitParameterization(boost::optional<double> path_tolerance,
+                                                                           boost::optional<double> resample_dt,
+                                                                           boost::optional<double> min_angle_change)
   : totg_(path_tolerance, resample_dt, min_angle_change)
 {
 }
 
 bool IterativeTorqueLimitParameterization::computeTimeStampsWithTorqueLimits(
     robot_trajectory::RobotTrajectory& trajectory, const geometry_msgs::Vector3& gravity_vector,
     const std::vector<geometry_msgs::Wrench>& external_link_wrenches, const std::vector<double>& joint_torque_limits,
-    double accel_limit_decrement_factor, const std::unordered_map<std::string, double>& velocity_limits,
-    const std::unordered_map<std::string, double>& acceleration_limits, const double max_velocity_scaling_factor,
-    const double max_acceleration_scaling_factor, const size_t max_iterations,
-    const bool reset_trajectory_after_max_iterations) const
+    const double max_velocity_scaling_factor, const double max_acceleration_scaling_factor,
+    boost::optional<double> accel_limit_decrement_factor, boost::optional<size_t> max_iterations,
+    boost::optional<bool> reset_trajectory_after_max_iterations) const
 {
   // 1. Call computeTimeStamps() to time-parameterize the trajectory with given vel/accel limits.
   // 2. Run forward dynamics to check if torque limits are violated at any waypoint.
   // 3. If a torque limit was violated, decrease the acceleration limit for that joint and go back to Step 1.
 
+  const double accel_decrement_factor = accel_limit_decrement_factor.get_value_or(0.1);
+  const size_t max_iter = max_iterations.get_value_or(10);
+  const bool reset_trajectory = reset_trajectory_after_max_iterations.get_value_or(false);
```

Comment:
nit: I'd still include part of the suffix to keep the semantic meaning
```suggestion
  const bool reset_trajectory_after_iter = reset_trajectory_after_max_iterations.get_value_or(false);
```